src/types.rs

```diff
@@ -483,22 +483,16 @@ impl Rewrite for ast::Ty {
                 ty.rewrite(context, budget, offset + 1).map(|ty_str| format!("({})", ty_str))
             }
             ast::TyTup(ref tup_ret) => {
-                let inner = try_opt!(tup_ret.iter()
-                                            .map(|item| item.rewrite(context, width, offset))
-                                            .fold(Some("".to_owned()),
-                                                  |sum, x| {
-                                                      match (sum, x) {
-                                                          (Some(sum), Some(x)) => {
-                                                              if sum == "" {
-                                                                  // First item.
-                                                                  Some(x)
-                                                              } else {
-                                                                  Some(sum + ", " + &x)
-                                                              }
-                                                          }
-                                                          _ => None,
-                                                      }
-                                                  }));
+                let inner = if let [ref item] = &**tup_ret {
+                    try_opt!(item.rewrite(context, width, offset).map(|rewrite| rewrite + ","))
+                } else {
+                    let rewrites: Option<Vec<_>>;
+                    rewrites = tup_ret.iter()
```

Comment:
The map(...).collect() will return an vector of optional string, not an optional vector of string. You're forcing it to ```Option<Vec<String>>``` by declaring the type earlier, so the next try_opt! will not do anything. This code will omit all items failed to format I believe.

The original code has a fold(...) part to detect if there is any None inside the vector, and returns None for the inner string if that's the case.